### ISSUE TITLE
Error basic::Singleton 6

### ISSUE THREAD

**Issue by AnantaYudica** [AnantaYudica/basic#49]
```
#include "Singleton.h"
#define USING_BASIC_TEST_MEMORY
#include "Test.h"

BasicTest& bt = BasicTest::ms_instance;

class Value
{
private:
    int m_value;
public:
    Value() : m_value(0) {}
    Value(const int& val) : m_value(val){}
public:
    int& GetValue() {return m_value;}
    const int& GetValue() const {return m_value;} 
};

class C :
    public Value,
    public basic::Singleton<C*>
{
    friend basic::Singleton<C*>;
private:
    static void Construct(void * p)
    {
        new (p) C*(new C);
    }
    static void Destroy(C** p)
    {
        delete *p;
    }
    C() {}
};


int main()
{
    auto x = C::GetInstance();
}

```

`Singleton.h:404:12: error: cannot dynamic_cast '((std::__shared_ptr_access<C*, (__gnu_cxx::_Lock_policy)2, false, false>*)(& _helper::_basic::_Singleton<C*>::ms_instance))->std::__shared_ptr_access<C*, (__gnu_cxx::_Lock_policy)2, false, false>::operator*()' (of type 'using element_type = class C* {aka class C*}') to type 'class C*&' (target is not pointer or reference to class)`

**Comment by AnantaYudica** [comment]
Fixed in v0.1-beta-2.

**Comment by AnantaYudica** [comment]
Reopen, not fixed in v0.1-beta-2, Related with bug #46
